cmd/loadtest/account.go

```diff
@@ -139,16 +142,36 @@ func NewAccountPool(ctx context.Context, client *ethclient.Client, fundingPrivat
 		return nil, fmt.Errorf("unable to get latestBlockNumber: %w", err)
 	}
 
+	if fundingAmount.Cmp(big.NewInt(0)) == 0 {
+		log.Info().
+			Msg("address-funding-amount is zero - account funding disabled")
+
+		// Create account pool without funding capability
+		return &AccountPool{
+			currentAccountIndex:   0,
+			client:                client,
+			accounts:              make([]Account, 0),
+			fundingPrivateKey:     fundingPrivateKey,
+			fundingAmount:         big.NewInt(0), // Keep as 0 to indicate no funding
+			chainID:               chainID,
+			accountsPositions:     make(map[common.Address]int),
+			latestBlockNumber:     latestBlockNumber,
+			clientRateLimiter:     rate.NewLimiter(rate.Every(50*time.Millisecond), 1),
+			fundingDisabledLogged: false,
+		}, nil
```

Comment:
to avoid code duplication, you can set the funding amount to zero in the if branch and then create the account pool (outside of the if branch) with the correct funding amount